src/components/autocomplete/js/autocompleteController.js

```diff
@@ -441,12 +441,21 @@ function MdAutocompleteCtrl ($scope, $element, $mdUtil, $mdConstant, $mdTheming,
         select(ctrl.index);
         break;
       case $mdConstant.KEY_CODE.ESCAPE:
+        // Only return if blur is disabled
```

Comment:
Can we refactor this into a helper function? Lots of logic to try to understand within a switch statement 😄